itk_wasm_env.bash

```diff
@@ -5,16 +5,16 @@ function die() {
     exit 1
 }
 
-if [$OSTYPE == "cygwin"] || [$OSTYPE == "msys"] || [$OSTYPE =="win32"]; then
+if test "$OSTYPE" = "cygwin" || test "$OSTYPE" = "msys" || test "$OSTYPE" = "win32"; then
     echo "Windows platform detected ... adding \"/Zc:__cplusplus /DNOMINMAX\" to \$CXXFLAGS"
     export CXXFLAGS="/Zc:__cplusplus /DNOMINMAX"
```

Comment:
My current script shows, but can't remember why I updated it. I will figure it out and post here again:
```bash
export CXXFLAGS="/Zc:__cplusplus /Zc:preprocessor /DNOMINMAX"
```

Reply:
Or I can just update it in a different PR if `/Zc:preprocessor` is a legit requirement for MSVC native builds.

Reply:
@jadh4v thanks! I added that flag.

I am going to merge to rebase a different branch on this, but if you test and find any issues with MSVC, another patch would be appreciated.